### PR TITLE
Make Sphinx documentation

### PR DESCRIPTION
**Pull request by veskil**
This PR contains documentation text files and Sphinx configuration files for automatically building documentation.

The documentation is intended for upload on readthedocs.io, and an example can be seen at: https://neffint.readthedocs.io/en/latest/

The docs can also be built locally by running:

    pip install .[docs]
    sphinx-build -b html docs/source docs/build/html

### CODE REVIEW

**Comment by veskil** [comment]
@giadarol I assume you're called giadarol on ReadTheDocs as well, so I can make you admin, but I didn't dare to do it before you confirm that is actually your name.

@nmounet If you also want to be an admin on ReadTheDocs you have to make an account there and tell me the username.

**Comment by nmounet** [comment]
great!

my username on ReadTheDocs: nimounet

File: docs/source/index.rst [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+   :maxdepth: 1
+   :caption: Contents:
+
+   theory
+   ex_fixed_grid_FI.ipynb
+   ex_adaptive.ipynb
+   api
+
+
+Installation procedure
```
Comment (nmounet) [inline docs/source/index.rst:23]: we could also provide the installation procedure for 1) compiling the doc, and 2) launching the tests

File: docs/source/theory.rst [hunk 2]
```diff
@@ -0,0 +1,235 @@
...
+the integrand :math:`\psi(\omega) e^{i \omega t}` oscillates with a very short period.
+This can cause problems when using a standard Fast Fourier Transform (FFT) technique,
+since it relies on the assumption that on, each frequency subinterval :math:`[\omega_k, \omega_{k+1}]` used in the computation,
+
+.. math::
+    \int_{\omega_k}^{\omega_{k+1}} \psi(\omega) e^{i \omega t} d\omega
+    \approx
+    (\omega_{k+1} - \omega_k) \psi(\omega_k) e^{i \omega t},
+
+which for high `t` require a very fine frequency spacing to be correct.
```
Comment (nmounet) [inline docs/source/theory.rst:23]: here we might want to cite the Numerical Recipes' book (see PhD)

File: docs/source/theory.rst [hunk 3]
```diff
@@ -0,0 +1,235 @@
...
+The Neffint Method
+------------------
+
+When computing Fourier integrals using the Neffint method, the integration range is first split into main integration range and asymptotic regions:
+
+.. math::
+    \int_{-\infty}^{\infty} \psi(\omega) e^{i \omega t} d\omega
+    \approx
+    \int_{-\infty}^{\omega_{min}} \psi(\omega) e^{i \omega t} d\omega
+    + \int_{-\omega_{min}}^{\omega_{max}} \psi(\omega) e^{i \omega t} d\omega
```
Comment (nmounet) [inline docs/source/theory.rst:39]: no minus sign in front of `\omega_{min}`

**Comment by giadarol** [comment]
> @giadarol I assume you're called giadarol on ReadTheDocs as well, so I can make you admin? I didn't dare to do it before you confirm that is actually your name.
> 
> @nmounet If you also want to be an admin on ReadTheDocs you have to make an account there and tell me the username.

I confirm :-)

File: docs/source/theory.rst [hunk 4]
```diff
@@ -0,0 +1,235 @@
...
+
+Introduction
+------------
+
+The Neffint package is mainly designed for with computing Fourier integrals, i.e.
+
+.. math::
+    \int_{-\infty}^{\infty} \psi(\omega) e^{i \omega t} d\omega,
+
+using a quadrature method largely based on section 1.6.3 in [N.Mounet-PhD]_, which builds on the work of [Filon]_.
```
Comment (nmounet) [inline docs/source/theory.rst:12]: we might want to cite as well the less detailed version of the method published [here](https://cds.cern.ch/record/2841806/files/document.pdf)

File: docs/source/theory.rst [hunk 5]
```diff
@@ -0,0 +1,235 @@
...
+Often, one deals with functions that are known to be real in time domain,
+in which case one only needs to integrate over half the frequency domain.
+One can in that case neglect the negative asymptotic term and set :math:`\omega_{min} = 0`
+(or arbitrarily close in the case of a singularity at `0`).
+
+
+Integrating the main integration range
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The main integration range we will split into subintervals :math:`[\omega_k, \omega_{k+1}]`,
```
Comment (nmounet) [inline docs/source/theory.rst:68]: `We split the main integration range into [...]`

File: docs/source/index.rst (hunk 1 above)
Comment (veskil) [reply docs/source/index.rst:23]: Will do

File: docs/source/theory.rst [hunk 6]
```diff
@@ -0,0 +1,235 @@
...
+
+In both solutions,
+
+.. math::
+    \Delta_k = \omega_{k+1} - \omega_k
+
+denotes the length of the frequency interval. It is also worth noting that while e.g. :math:`p_k(\omega)` is written in the expressions,
+by construction, this is identical to :math:`\psi(\omega)` as well.
+There is in practice therefore not necessary to compute the interpolant (i.e. slopes and constants on each interval for linear interpolation),
+as knowing the function values is enough.
```
Comment (nmounet) [inline docs/source/theory.rst:163]: I don't get what is meant here. First, `p_k` is not exactly `\psi` (it is its approximation), and second, you need the slopes in the PCHIP case.

**Comment by veskil** [comment]
> @giadarol I assume you're called giadarol on ReadTheDocs as well, so I can make you admin? I didn't dare to do it before you confirm that is actually your name.
> 
> @nmounet If you also want to be an admin on ReadTheDocs you have to make an account there and tell me the username.

You are both added now

File: docs/source/theory.rst (hunk 6 above)
Comment (veskil) [reply docs/source/theory.rst:165]: What I meant is that you don't actually need to compute e.g. the coefficients of the polynomial, that knowing the function values (and slopes for PCHIP) is sufficient. But I agree it is not clear, I will rephrase

**Comment by veskil** [comment]
I added some links from the documentation to the Github page and vice versa.

All your changes look good to me

**Comment by nmounet** [comment]
excellent and clean doc!

**Review by nmounet — APPROVED** [review]

**Pull request merged**